src/cast_core/src/CollectionScanner.cpp

```diff
@@ -209,6 +212,30 @@ void collectionScan(CollectionScanner::PhaseConfig* config,
                     scanFinished = true;
                     break;
                 }
+                // tetsuo-cpp: We don't want to do this after every insert.
+                // Figure out a better spot to put this.
+                if (config->scanRate) {
+                    const auto now = SteadyClock::now();
+
+                    // Calculate the scan rate as a double for ease of use.
+                    const auto spec = config->scanRate->getBaseSpec();
+                    assert(spec);
+                    const auto secs = spec->per.count() / (1000 * 1000 * 1000);
+                    const auto elapsed = (now - start).count() / (1000 * 1000 * 1000);
+                    const auto expectedSize =
+                        (static_cast<double>(elapsed) / secs) * spec->operations;
```

Comment:
I'm overloading ops/sec to mean bytes/sec. I need to add a bytes/sec variant to `RateSpec`.

- [ ] Add a bytes/second rate to the `RateSpec` implementation.